app/models/labware.rb

```diff
@@ -70,11 +70,29 @@ def receptacle_proxy
   has_many :creation_batches, class_name: 'Batch', through: :requests_as_target, source: :batch
 
   belongs_to :purpose, foreign_key: :plate_purpose_id, optional: true, inverse_of: :labware
-  has_one :spiked_in_buffer_links,
-          -> { joins(:ancestor).where(labware: { sti_type: 'SpikedBuffer' }).direct },
-          class_name: 'AssetLink',
-          foreign_key: :descendant_id,
-          inverse_of: :descendant
+
+  # Annoyingly we need two versions of the association, as Rails 5 and Rails 6 seem to
+  # have distinct behaviour differences.
+  if Rails.version.starts_with?('5')
+    has_one :spiked_in_buffer_links,
+            # If we try to use the Rails 6 version, Rails 5 doesn't seem to perform the join
+            -> { joins(:ancestor).where(labware: { sti_type: 'SpikedBuffer' }).direct },
+            class_name: 'AssetLink',
+            foreign_key: :descendant_id,
+            inverse_of: :descendant
+  else
+    has_one :spiked_in_buffer_links,
+            -> {
+              # If we use joins(:ancestor) here, Rails 6 seems to try to find an ancestor
+              # relationship on Labware.
+              joins('INNER JOIN labware AS sbl ON sbl.id = asset_links.ancestor_id')
+                .where(sbl: { sti_type: 'SpikedBuffer' })
+                .direct
+            },
+            class_name: 'AssetLink',
+            foreign_key: :descendant_id,
+            inverse_of: :descendant
+  end
```

Comment:
This one is a little unfortunate. As far as I'm concerned, both versions should work with either version of rails. However, each falls foul of strange version specific behaviour.

Reply:
Ahh, found a universal approach that's also a lot nice than my previous attempt.